javascripts/discourse/components/discourse-tag-banners.hbs

```diff
@@ -6,19 +6,24 @@
       {{did-update this.getTagInfo this.shouldRender}}
       {{will-destroy this.resetTag}}
     >
-      {{#if this.categoryBannerPresence.isPresent}}
-        <DiscourseTagBannersTextOnly
+      {{#if
+        (or
+          (not this.categoryBannerPresence.isPresent)
+          (theme-setting "show_with_category_banners")
+        )
+      }}
```

Comment:
Show category banner and tag banner if theme-setting is checked, regardless of presence. Or statement, if one is true, valid statement.